pkg/util/cgroup/cgroup_cpu_linux.go

```diff
@@ -66,10 +66,30 @@ func inContainer(path string) bool {
 	if err != nil {
 		return false
 	}
-	if strings.Contains(string(v), "docker") ||
-		strings.Contains(string(v), "kubepods") ||
-		strings.Contains(string(v), "containerd") {
-		return true
+
+	// For cgroup V1, check /proc/self/cgroup
+	if path == procPathCGroup {
+		if strings.Contains(string(v), "docker") ||
+			strings.Contains(string(v), "kubepods") ||
+			strings.Contains(string(v), "containerd") {
+			return true
+		}
+	}
+
+	// For cgroup V2, check /proc/self/mountinfo
+	if path == procPathMountInfo {
```

Comment:
what's the case you are trying to handle? please show the OS, container hypervisor and the content of `/proc/self/cgroup` and `/proc/self/mountinfo` in your case